task24/index.html

```diff
@@ -0,0 +1,243 @@
+<!DOCTYPE html>
+<html lang="ja">
+  <head>
+    <meta charset="UTF-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
+    />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <script type="module" src="js/main.js"></script>
+    <title>JS課題24</title>
+  </head>
+  <body>
+    <section class="flex items-center justify-center min-h-screen">
+      <form action="post w-96">
+        <h1 class="mb-8 text-3xl text-center">👋 Welcome!</h1>
+        <div class="my-4">
+          <label for="e-mail" class=" mt-1">ユーザー名
+            <input
+              id="js-user"
+              type="name"
+              name="name"
+              placeholder="user-name"
+              class="block border border-solid border-gray-500 rounded-md w-80 p-1"
+              required
+            />
+          </label>
+        </div>
+        <div class="my-4">
+          <label for="e-mail" class="mt-1">メールアドレス
+            <input
+              id="js-getMailAdress"
+              type="email"
+              name="email"
+              placeholder="sample@sample.com"
+              class="block border border-solid border-gray-500 rounded-md w-80 p-1"
+              required
+            />
+          </label>
+        </div>
+        <div class="my-4">
+          <label for="password" class="mt-1">パスワード
+            <input
+              id="js-getPassWord"
+              type="password"
+              placeholder="英数字8文字以上"
+              class="block border border-solid border-gray-500 rounded-md w-80 p-1"
+              pattern="^[a-zA-Z0-9]+$"
+              minlength="8"  maxlength="12" 
+              required
+            />
+          </label>
+        </div>
+        <div class="my-4">
+          <input id="js-checkBox" type="checkbox" value="同意する" class="" disabled />
+          <label for="check" class=""
+            ><a
+              id="js-modalPlace"
+              href="#js-modalPlace"
+              class="ml-1 text-blue-800 underline"
+              >利用規約</a
+            >を読み同意してください</label
+          >
+        </div>
+        <div class="my-8 mx-auto text-center w-52 rounded-md bg-yellow-500 p-3 login-button">
+          <a href="" id="js-login" class="block text-white">ログイン</a>
+        </div>
+      </form>
+    </section>
+    <div id="js-modalContents" class="hidden fixed inset-0 flex justify-center items-center z-50 bg-gray-700 bg-opacity-95">
+      <div id="js-modal" class="max-w-screen-sm h-96 p-4 bg-white overflow-y-auto rounded-lg">
+        <h1>利用規約</h1>
+        <p>
+          この利用規約（以下，「本規約」といいます。）は，＿＿＿＿＿（以下，「当社」といいます。）がこのウェブサイト上で提供するサービス（以下，「本サービス」といいます。）の利用条件を定めるものです。登録ユーザーの皆さま（以下，「ユーザー」といいます。）には，本規約に従って，本サービスをご利用いただきます。
+        </p>
```

Comment:
IMO
I also learned from this assignment that from an accessibility point of view, when creating a modal with a div, it should have ``role="dialog"``. 

[Review in my assignment](https://github.com/csakyo/homework/pull/38#discussion_r922006437)
[Ref](https://developer.mozilla.org/ja/docs/Web/Accessibility/ARIA/Roles/dialog_role#%E8%AA%AC%E6%98%8E)

Reply:
Thanks for sharing. I had never heard of this before and learned a lot. Is this implementation correct?
[240060e](https://github.com/seeeen044/JavaScript_handson/pull/26/commits/240060e5fb642c5e02717055c64db332e777ebe6)